src/main/java/org/primefaces/model/timeline/TimelineGroup.java

```diff
@@ -44,6 +45,17 @@
      */
     private String title;
 
+    /**
+     * root level of the nested groups.
+     */
+    private Integer treeLevel;
+
+    /**
+     *
+     * it is a set of ids of nested groups that a group or nested group contains.
+     */
+    private HashSet<String> nestedGroups;
```

Comment:
This should just be Set and not HashSet.  That way someone could use a TreeSet or any other kind of Set they want.